auth/tests/api/users.rs

```diff
@@ -222,6 +231,70 @@ mod needs_docker {
         );
     }
 
+    #[tokio::test]
+    async fn insert_and_increment_and_delete_rds_subscription() {
+        let app = app().await;
+
+        // Create user first so one exists in the database.
+        let response = app.post_user("test-user", "basic").await;
+        assert_eq!(response.status(), StatusCode::OK);
+
+        // Extract the API key from the response so we can use it in a future request.
+        let body = hyper::body::to_bytes(response.into_body()).await.unwrap();
+        let user: Value = serde_json::from_slice(&body).unwrap();
+        let basic_user_key = user["key"].as_str().unwrap();
+
+        // Make sure JWT does not allow any RDS instances
+        let claim = app.get_claim(basic_user_key).await;
+        assert_eq!(claim.sub, "test-user");
+        assert_eq!(claim.limits.rds_quota, 0);
+
+        // Send a request to insert an RDS subscription for the test user.
+        let response = app
+            .post_subscription("test-user", "sub_Eoarshy23pointInira", "rds", 1)
+            .await;
+        assert_eq!(response.status(), StatusCode::OK);
+
+        // Fetch the user and verify they have an rds subscription.
+        let response = app.get_user_typed("test-user").await;
+
+        assert_eq!(
+            response.subscriptions.len(),
+            1,
+            "there should be one subscription"
+        );
+        assert_eq!(
+            response.subscriptions[0].r#type,
+            models::user::SubscriptionType::Rds
+        );
+        assert_eq!(response.subscriptions[0].quantity, 1);
+
+        // Make sure JWT has the quota
+        let claim = app.get_claim(basic_user_key).await;
+        assert_eq!(claim.limits.rds_quota, 1);
+
+        // Send another request to insert an RDS subscription for the user.
+        // This uses a different subscription id to make sure we only keep record of one
```

Comment:
I think what's guaranteeing we only keep one is that this account already has a subscription of the RDS type, right? It could be the same subscription ID, which I imagine will be the case if you increase the quantity of the subscription in the console (I'm not up to speed on the implementation in the console, but I assume increasing the quantity doesn't change the subscription id)?